### DISCUSSION TITLE
How to access inject/autowire depedancies in Job

### DISCUSSION THREAD

**Discussion by laz001001** [jobrunr/jobrunr#754] [Q&A]
Hello :-) 

I'm a relative newbie when it comes to Spring - so looking for some guideance:

So I have a method that kicks off a job:

   
    @Autowired
    Recalculator R;

    @Autowired
    SampleJobService SJS;

    @Autowired 
    private JobScheduler jobScheduler;

    public void Create() {

        //some logic

        **jobScheduler.enqueue(
                () -> SJS.RecalculateJob());**

        //some logic

    }

When i debug this, my Recalculator R dependancy is fully resovled:

The Recalcuate job looks like:

```
@Component 
public class SampleJobService {

    @Autowired
    Recalculator R;

    //some other stuff/repositories

    public void RecalculateJob()
    {

        //some logic

        R.Recalculate("some params");

        //some logic
    }
}
```

But - when i debug here, my Recalculator R and other dependancies are null.  I've looked over all the other discussions, emailed the developer looked throughout Youtube, i cannot get a solution. How do I do this?

Thanks,

Laz

**Comment by rdehuyss** [comment]
Are you using the `jobrunr-spring-boot-x-starter`? Or, how have you setup JobRunr?
